src/notifications/notifications.service.ts

```diff
@@ -0,0 +1,46 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Image } from 'src/entity/image.entity';
+import { Notification } from 'src/entity/notification.entity';
+import { User } from 'src/entity/user.entity';
+import { sendImageEmailNotification } from 'src/helpers/email.helper';
+import { Repository } from 'typeorm';
+
+@Injectable()
+export class NotificationsService {
+  constructor(
+    @InjectRepository(User)
+    private readonly usersRepository: Repository<User>,
+    @InjectRepository(Notification)
+    private readonly notificationRepository: Repository<Notification>,
+  ) {}
+
+  async imageUploaded(image: Image) {
+    {
+      try {
+        await this.notificationRepository.save({
+          type: 'image',
+          user: image.user,
+          message: '',
+          meta: JSON.stringify({ imageId: image.id }),
+        });
+      } catch (err)
+      { 
+        return BadRequestException;
```

Comment:
This is not good - we don't want to return here, because as we said the notification is a side-effect and errors happened in relation with them are not errors about which user should be informed. And why did you use BadRequestException  anyway? In this case it would be the best if you just logged the error and that's enough.